### PR TITLE
[WIP] Update bug fixes

### PR DESCRIPTION
**Pull request by SamTov**
Fix some bugs in the RL code.

* Removed self-sensing from particle sensing
* Added test for initialisation to be sure that multiple initialisations do not change the state as it should not.
* Checked that there is no cyclic changing of observables, i.e, fixed really dumb bug I wrote into the code the first time around.

### CODE REVIEW

**Review by christophlohrmann — COMMENTED** [review]
seems reasonable. Can you update the tests to reflect the changes?

**Comment by SamTov** [comment]
> seems reasonable. Can you update the tests to reflect the changes?

Yep! I will fix the tests and then it should be okay for merge.

**Comment by SamTov** [comment]
@christophlohrmann I have added tests to the observable and task that ensure that when the observable is called many times without changes that it also does not change its value.

**Review by christophlohrmann — DISMISSED** [review]

**Comment by christophlohrmann** [comment]
maybe @Dalouvid should have a look at this too before we merge

**Comment by SamTov** [comment]
> maybe @Dalouvid should have a look at this too before we merge

Sounds good! There is no rush.

File: swarmrl/tasks/searching/species_search.py
Comment (Dalouvid) [inline swarmrl/tasks/searching/species_search.py:1]: "Observable" in the task is kinda misleading and could cause confusion.

File: swarmrl/observables/particle_sensing.py
Comment (Dalouvid) [inline swarmrl/observables/particle_sensing.py:88]: The variable name "test_points" was not clear to me. One could think about renaming that to a more intuitive name. Something like "sensing_points", "sensor_points" or "sensed_points"

**Review by Dalouvid — CHANGES_REQUESTED** [review]
Nice job. Except for the mentioned point, super clean.

File: swarmrl/observables/particle_sensing.py
Comment (SamTov) [reply swarmrl/observables/particle_sensing.py]: Resolved in https://github.com/SwarmRL/SwarmRL/pull/20/commits/63ab2e0d3b0653d9db8cfd473644b14e7a887a3c

File: swarmrl/tasks/searching/species_search.py
Comment (SamTov) [reply swarmrl/tasks/searching/species_search.py:1]: Resolved in https://github.com/SwarmRL/SwarmRL/pull/20/commits/63ab2e0d3b0653d9db8cfd473644b14e7a887a3c

**Review by Dalouvid — APPROVED** [review]
Looks good!

**Pull request merged**
